main.go

```diff
@@ -85,38 +85,61 @@ func main() {
 		log.Fatalf(usage)
 	}
 
-	url, err := url.Parse(args[0])
+	uri := schemify(args[0])
```

Comment:
Please break this out into a function
```go
url := parseURL(args[0])
```
Then handle the error condition inside `parseURL`.